lib/src/facts/linux/networking_resolver.cc

```diff
@@ -12,9 +14,18 @@ using namespace std;
 using namespace facter::util::posix;
 
 namespace lth_file = leatherman::file_util;
+namespace lth_exe  = leatherman::execution;
 
 namespace facter { namespace facts { namespace linux {
 
+    networking_resolver::data networking_resolver::collect_data(collection& facts)
+    {
+        read_routing_table();
```

Comment:
Yeah. In the `get_primary_interface` method below I check to see if we read the routing table using `ip` here so that we don't re-read it from proc and duplicate work/logic. I could have this method return the vector and just set a bool on the class instance, but just storing the vector of routes as a class members seemed easier seemed easier.